packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -470,24 +574,43 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
     })
     this.socketService.on(
       SocketActionTypes.CREATE_NETWORK,
-      async (communityId: string, callback: (response?: NetworkInfo) => void) => {
+      async (communityId: string, callback: (response: NetworkInfo | undefined) => void) => {
         this.logger(`socketService - ${SocketActionTypes.CREATE_NETWORK}`)
         callback(await this.createNetwork(communityId))
       }
     )
-    this.socketService.on(SocketActionTypes.CREATE_COMMUNITY, async (args: InitCommunityPayload) => {
-      await this.createCommunity(args)
-    })
-    this.socketService.on(SocketActionTypes.LAUNCH_COMMUNITY, async (args: InitCommunityPayload) => {
-      this.logger(`socketService - ${SocketActionTypes.LAUNCH_COMMUNITY}`)
-      if ([ServiceState.LAUNCHING, ServiceState.LAUNCHED].includes(this.communityState)) return
-      this.communityState = ServiceState.LAUNCHING
-      await this.launchCommunity(args)
-    })
+    this.socketService.on(
+      SocketActionTypes.CREATE_COMMUNITY,
+      async (args: InitCommunityPayload, callback: (response: Community | undefined) => void) => {
+        this.logger(`socketService - ${SocketActionTypes.CREATE_COMMUNITY}`)
+        callback(await this.createCommunity(args))
+      }
+    )
+    // TODO: Rename to JOIN_COMMUNITY?
+    this.socketService.on(
+      SocketActionTypes.LAUNCH_COMMUNITY,
+      async (args: InitCommunityPayload, callback: (response: Community | undefined) => void) => {
+        this.logger(`socketService - ${SocketActionTypes.LAUNCH_COMMUNITY}`)
+        callback(await this.joinCommunity(args))
+      }
+    )
+    // TODO: With the Community model on the backend, there is no need to call
+    // SET_COMMUNITY_METADATA anymore. We can call updateCommunityMetadata when
+    // creating the community.
     this.socketService.on(
       SocketActionTypes.SET_COMMUNITY_METADATA,
-      async (payload: CommunityMetadata, callback: (response?: CommunityMetadata) => void) => {
-        const meta = await this.storageService?.updateCommunityMetadata(payload)
+      async (payload: CommunityMetadata, callback: (response: CommunityMetadata | undefined) => void) => {
+        const meta = await this.storageService.updateCommunityMetadata(payload)
```

Comment:
I see that you added throwing exception in updateCommunityMetadata. Maybe we should catch it here?

Reply:
Where is the exception? I think I'm missing it.